### ISSUE TITLE
[BUG] JobRunr is not compatible with Mongo DB Driver 4.1 (default Mongo driver in Spring Boot 2.3)

### ISSUE THREAD

**Issue by rdehuyss** [jobrunr/jobrunr#55]
**Describe the bug**

Background Job Server is not starting when using Spring Boot 2.3

**Environment**
I'm using version 1.1.0
I'm using the following StorageProvider MongoDBStorageProvider 

**To Reproduce**
Run tests with latest MongoDB Driver

**Expected behavior**
Tests should pass

**Extra information**
```
Exception in thread "extShutdownHook" org.bson.codecs.configuration.CodecConfigurationException: The uuidRepresentation has not been specified, so the UUID cannot be encoded.
	at org.bson.codecs.UuidCodec.encode(UuidCodec.java:72)
	at org.bson.codecs.UuidCodec.encode(UuidCodec.java:37)
	at com.mongodb.client.model.BuildersHelper.encodeValue(BuildersHelper.java:37)
	at com.mongodb.client.model.Filters$SimpleEncodingFilter.toBsonDocument(Filters.java:1175)
	at com.mongodb.internal.operation.Operations.toBsonDocument(Operations.java:514)
	at com.mongodb.internal.operation.Operations.bulkWrite(Operations.java:406)
	at com.mongodb.internal.operation.Operations.deleteOne(Operations.java:319)
	at com.mongodb.internal.operation.SyncOperations.deleteOne(SyncOperations.java:173)
	at com.mongodb.client.internal.MongoCollectionImpl.executeDelete(MongoCollectionImpl.java:982)
	at com.mongodb.client.internal.MongoCollectionImpl.deleteOne(MongoCollectionImpl.java:507)
	at com.mongodb.client.internal.MongoCollectionImpl.deleteOne(MongoCollectionImpl.java:502)
	at org.jobrunr.storage.nosql.mongo.MongoDBStorageProvider.signalBackgroundJobServerStopped(MongoDBStorageProvider.java:137)
	at org.jobrunr.storage.ThreadSafeStorageProvider.signalBackgroundJobServerStopped(ThreadSafeStorageProvider.java:53)
	at org.jobrunr.server.ServerZooKeeper.stop(ServerZooKeeper.java:52)
	at org.jobrunr.server.BackgroundJobServer.stopZooKeepers(BackgroundJobServer.java:187)
	at org.jobrunr.server.BackgroundJobServer.stop(BackgroundJobServer.java:110)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by rdehuyss** [comment]
Relates to https://github.com/spring-projects/spring-boot/issues/22321

**Comment by rdehuyss** [comment]
Solved with release v1.1.1
